src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -0,0 +1,321 @@
+            }
+            
+
+            return FormValidation.ok();
+        };
+        
+        public FormValidation doCheckModelCoverageChkBx(@QueryParameter boolean modelCoverageChkBx) {
+            List<Function<String, FormValidation>> listOfCheckMethods =
+                    new ArrayList<Function<String, FormValidation>>();
+            if (modelCoverageChkBx) {
+                listOfCheckMethods.add(chkModelCoverageSupport);
+            }
+            return getFirstErrorOrWarning(listOfCheckMethods);
+        }
+        
+        Function<String, FormValidation> chkModelCoverageSupport = (String matlabRoot) -> {
+            FilePath matlabRootPath = new FilePath(new File(matlabRoot));
+            rel = new MatlabReleaseInfo(matlabRootPath);
+            final MatrixPatternResolver resolver = new MatrixPatternResolver(matlabRoot);
+            if(!resolver.hasVariablePattern()) {
+                try {
+                    if (rel.verLessThan(MatlabBuilderConstants.BASE_MATLAB_VERSION_MODELCOVERAGE_SUPPORT)) {
+                        return FormValidation
+                                .warning(Message.getValue("Builder.matlab.modelcoverage.support.warning"));
+                    }
+                } catch (MatlabVersionNotFoundException e) {
+                    return FormValidation.warning(Message.getValue("Builder.invalid.matlab.root.warning"));
+                }
+            }
+            
+            
+            return FormValidation.ok();
+        };
+        
+        public FormValidation doCheckStmResultsChkBx(@QueryParameter boolean stmResultsChkBx) {
+            List<Function<String, FormValidation>> listOfCheckMethods =
+                    new ArrayList<Function<String, FormValidation>>();
+            if (stmResultsChkBx) {
+                listOfCheckMethods.add(chkSTMResultsSupport);
+            }
+            return getFirstErrorOrWarning(listOfCheckMethods);
+        }
+        
+        Function<String, FormValidation> chkSTMResultsSupport = (String matlabRoot) -> {
+            FilePath matlabRootPath = new FilePath(new File(matlabRoot));
+            rel = new MatlabReleaseInfo(matlabRootPath);
+            final MatrixPatternResolver resolver = new MatrixPatternResolver(matlabRoot);
+            if(!resolver.hasVariablePattern()) {
+                try {
+                    if (rel.verLessThan(MatlabBuilderConstants.BASE_MATLAB_VERSION_EXPORTSTMRESULTS_SUPPORT)) {
+                        return FormValidation
+                                .warning(Message.getValue("Builder.matlab.exportstmresults.support.warning"));
+                    }
+                } catch (MatlabVersionNotFoundException e) {
+                    return FormValidation.warning(Message.getValue("Builder.invalid.matlab.root.warning"));
+                }
+            }
+            
+            
+            return FormValidation.ok();
+        };
+        
+        public FormValidation getFirstErrorOrWarning(

```

Comment:
There are 2 different definitions of getFirstErrorOrWarning. Should this be in a common place? And why would the implementations differ a bit?

Reply:
As discussed. moved the implementation of getFirstErrorOrWarning to new utility class. and made it consistent to take only one argument that is lambda function. i tried making matlabRoot as class property and passing it to all the methods however i was facing an  issue with the same. there seem to be an issue with lambda functions if unhanddled exception occurs https://github.com/google/guice/issues/757 it works when i keep matlabroot under try catch.